compiler/src/main/kotlin/software/amazon/lastmile/kotlin/inject/anvil/processor/GenerateKotlinInjectComponentProcessor.kt

```diff
@@ -115,12 +119,23 @@ internal class GenerateKotlinInjectComponentProcessor(
 
         val isInterface = clazz.classKind == ClassKind.INTERFACE
         val parameters = clazz.primaryConstructor?.parameters ?: emptyList()
+        val componentAnnotations = mutableMapOf<String, TypeName>()
         val parametersAsSpec = parameters.map {
             ParameterSpec
                 .builder(
                     name = it.requireName(),
                     type = it.type.toTypeName(),
-                )
+                ).apply {
+                    it.annotations.forEach { annotation ->
+                        addAnnotation(annotation.toAnnotationSpec())
+                        if (annotation.isKotlinInjectComponentAnnotation()) {
+                            require(it.type.resolve().declaration.isOpen()) {
+                                "Component parameter must be open"
```

Comment:
Why is this required? The generated component could use a different parameter name.

Reply:
To be clear, I tested this and it works:
```kotlin
@Component
interface TestComponent {
    @Provides
    fun provideString(): String = "Hello"
}

abstract class SuperType(
    @Component val testComponent: TestComponent,
) {
    abstract val string: String
}

@Component
abstract class FinalComponent(
    @Component val testComponentDelegate: TestComponent,
) : SuperType(testComponentDelegate
```